Upgrade to syn 2

Errors when switching to syn 2
```rust
error[E0432]: unresolved import `syn::NestedMeta`
  --> butane_core/src/codegen/mod.rs:11:20
   |
11 |     MetaNameValue, NestedMeta,
   |                    ^^^^^^^^^^ no `NestedMeta` in the root

error[E0615]: attempted to take value of method `path` on type `&Attribute`
   --> butane_core/src/codegen/mod.rs:214:24
    |
214 |         .filter(|a| !a.path.is_ident("table"))
    |                        ^^^^ method, not a field
    |
help: use parentheses to call the method
    |
214 |         .filter(|a| !a.path().is_ident("table"))
    |                            ++

error[E0599]: no method named `parse_meta` found for reference `&Attribute` in the current scope
   --> butane_core/src/codegen/mod.rs:225:20
    |
225 |         })) = attr.parse_meta()
    |                    ^^^^^^^^^^ help: there is a method with a similar name: `parse_nested_meta`

error[E0026]: struct `MetaNameValue` does not have a field named `lit`
   --> butane_core/src/codegen/mod.rs:223:13
    |
223 |             lit: Lit::Str(s),
    |             ^^^ struct `MetaNameValue` does not have this field

error[E0615]: attempted to take value of method `path` on type `&Attribute`
   --> butane_core/src/codegen/mod.rs:242:24
    |
242 |                     !a.path.is_ident("pk")
    |                        ^^^^ method, not a field
    |
help: use parentheses to call the method
    |
242 |                     !a.path().is_ident("pk")
    |                            ++

error[E0615]: attempted to take value of method `path` on type `&Attribute`
   --> butane_core/src/codegen/mod.rs:243:31
    |
243 |                         && !a.path.is_ident("auto")
    |                               ^^^^ method, not a field
    |
help: use parentheses to call the method
    |
243 |                         && !a.path().is_ident("auto")
    |                                   ++

error[E0615]: attempted to take value of method `path` on type `&Attribute`
   --> butane_core/src/codegen/mod.rs:244:31
    |
244 |                         && !a.path.is_ident("sqltype")
    |                               ^^^^ method, not a field
    |
help: use parentheses to call the method
    |
244 |                         && !a.path().is_ident("sqltype")
    |                                   ++

error[E0615]: attempted to take value of method `path` on type `&Attribute`
   --> butane_core/src/codegen/mod.rs:245:31
    |
245 |                         && !a.path.is_ident("default")
    |                               ^^^^ method, not a field
    |
help: use parentheses to call the method
    |
245 |                         && !a.path().is_ident("default")
    |                                   ++

error[E0615]: attempted to take value of method `path` on type `&Attribute`
   --> butane_core/src/codegen/mod.rs:246:31
    |
246 |                         && !a.path.is_ident("unique")
    |                               ^^^^ method, not a field
    |
help: use parentheses to call the method
    |
246 |                         && !a.path().is_ident("unique")
    |                                   ++

error[E0615]: attempted to take value of method `path` on type `&Attribute`
   --> butane_core/src/codegen/mod.rs:279:68
    |
279 |         fields(ast_struct).find(|f| f.attrs.iter().any(|attr| attr.path.is_ident("pk")));
    |                                                                    ^^^^ method, not a field
    |
help: use parentheses to call the method
    |
279 |         fields(ast_struct).find(|f| f.attrs.iter().any(|attr| attr.path().is_ident("pk")));
    |                                                                        ++

error[E0615]: attempted to take value of method `path` on type `&Attribute`
   --> butane_core/src/codegen/mod.rs:291:40
    |
291 |     field.attrs.iter().any(|attr| attr.path.is_ident("auto"))
    |                                        ^^^^ method, not a field
    |
help: use parentheses to call the method
    |
291 |     field.attrs.iter().any(|attr| attr.path().is_ident("auto"))
    |                                            ++

error[E0615]: attempted to take value of method `path` on type `&Attribute`
   --> butane_core/src/codegen/mod.rs:295:40
    |
295 |     field.attrs.iter().any(|attr| attr.path.is_ident("unique"))
    |                                        ^^^^ method, not a field
    |
help: use parentheses to call the method
    |
295 |     field.attrs.iter().any(|attr| attr.path().is_ident("unique"))
    |                                            ++

error[E0615]: attempted to take value of method `path` on type `&&Attribute`
   --> butane_core/src/codegen/mod.rs:393:27
    |
393 |         .find(|attr| attr.path.is_ident("default"));
    |                           ^^^^ method, not a field
    |
help: use parentheses to call the method
    |
393 |         .find(|attr| attr.path().is_ident("default"));
    |                               ++

error[E0599]: no method named `parse_meta` found for reference `&Attribute` in the current scope
   --> butane_core/src/codegen/mod.rs:396:34
    |
396 |         Some(attr) => match attr.parse_meta() {
    |                                  ^^^^^^^^^^ help: there is a method with a similar name: `parse_nested_meta`

error[E0609]: no field `lit` on type `MetaNameValue`
   --> butane_core/src/codegen/mod.rs:397:47
    |
397 |             Ok(Meta::NameValue(meta)) => meta.lit,
    |                                               ^^^ unknown field
    |
    = note: available fields are: `path`, `eq_token`, `value`

error[E0599]: no method named `parse_meta` found for reference `&Attribute` in the current scope
   --> butane_core/src/codegen/mod.rs:506:42
    |
506 |         if let Ok(Meta::List(ml)) = attr.parse_meta() {
    |                                          ^^^^^^^^^^ help: there is a method with a similar name: `parse_nested_meta`

error[E0609]: no field `nested` on type `MetaList`
   --> butane_core/src/codegen/mod.rs:508:23
    |
508 |                 && ml.nested.iter().any(|nm| match nm {
    |                       ^^^^^^ unknown field
    |
    = note: available fields are: `path`, `delimiter`, `tokens`

```

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -35,7 +35,7 @@ rand = "0.8"
 rusqlite = "0.29"
 serde = { version = "1.0", default-features = false }
 serde_json = "1.0"
-syn = { version = "1.0", features = ["extra-traits", "full"] }
+syn = { version = "2", features = ["extra-traits", "full"] }
 uuid = "1.2"
 
 [workspace.metadata.release]

--- a/butane/tests/basic.rs
+++ b/butane/tests/basic.rs
@@ -3,14 +3,16 @@ use butane::{butane_type, find, model, query};
 use butane::{colname, prelude::*};
 use butane::{ForeignKey, ObjectState};
 use chrono::{naive::NaiveDateTime, offset::Utc, DateTime};
+use serde::Serialize;
 
 use butane_test_helper::*;
 
 #[butane_type]
 pub type Whatsit = String;
 
+// Note, Serialize derive exists solely to exercise the logic in butane_core::codegen::has_derive_serialize
 #[model]
-#[derive(PartialEq, Debug, Clone)]
+#[derive(PartialEq, Debug, Clone, Serialize)]
 struct Foo {
     id: i64,
     bam: f64,

--- a/butane_core/src/codegen/mod.rs
+++ b/butane_core/src/codegen/mod.rs
@@ -8,7 +8,7 @@ use regex::Regex;
 use syn::parse_quote;
 use syn::{
     punctuated::Punctuated, Attribute, Field, ItemEnum, ItemStruct, ItemType, Lit, LitStr, Meta,
-    MetaNameValue, NestedMeta,
+    MetaNameValue,
 };
 
 #[macro_export]
@@ -211,18 +211,21 @@ fn filter_helper_attributes(ast_struct: &ItemStruct) -> Vec<Attribute> {
         .attrs
         .clone()
         .into_iter()
-        .filter(|a| !a.path.is_ident("table"))
+        .filter(|a| !a.path().is_ident("table"))
         .collect()
 }
 
 fn config_from_attributes(ast_struct: &ItemStruct) -> dbobj::Config {
     let mut config = dbobj::Config::default();
     for attr in &ast_struct.attrs {
-        if let Ok(Meta::NameValue(MetaNameValue {
+        // #[table = "name"]
+        if let Meta::NameValue(MetaNameValue {
             path,
-            lit: Lit::Str(s),
+            value: syn::Expr::Lit(syn::ExprLit {
+                lit: Lit::Str(s), ..
+            }),
             ..
-        })) = attr.parse_meta()
+        }) = &attr.meta
         {
             if path.is_ident("table") {
                 config.table_name = Some(s.value())
@@ -239,11 +242,11 @@ fn remove_helper_field_attributes(
         syn::Fields::Named(fields) => {
             for field in &mut fields.named {
                 field.attrs.retain(|a| {
-                    !a.path.is_ident("pk")
-                        && !a.path.is_ident("auto")
-                        && !a.path.is_ident("sqltype")
-                        && !a.path.is_ident("default")
-                        && !a.path.is_ident("unique")
+                    !a.path().is_ident("pk")
+                        && !a.path().is_ident("auto")
+                        && !a.path().is_ident("sqltype")
+                        && !a.path().is_ident("default")
+                        && !a.path().is_ident("unique")
                 });
             }
             Ok(fields)
@@ -276,7 +279,7 @@ fn remove_existing_state_field(
 
 fn pk_field(ast_struct: &ItemStruct) -> Option<Field> {
     let pk_by_attribute =
-        fields(ast_struct).find(|f| f.attrs.iter().any(|attr| attr.path.is_ident("pk")));
+        fields(ast_struct).find(|f| f.attrs.iter().any(|attr| attr.path().is_ident("pk")));
     if let Some(id_field) = pk_by_attribute {
         return Some(id_field.clone());
     }
@@ -288,11 +291,14 @@ fn pk_field(ast_struct: &ItemStruct) -> Option<Field> {
 }
 
 fn is_auto(field: &Field) -> bool {
-    field.attrs.iter().any(|attr| attr.path.is_ident("auto"))
+    field.attrs.iter().any(|attr| attr.path().is_ident("auto"))
 }
 
 fn is_unique(field: &Field) -> bool {
-    field.attrs.iter().any(|attr| attr.path.is_ident("unique"))
+    field
+        .attrs
+        .iter()
+        .any(|attr| attr.path().is_ident("unique"))
 }
 
 fn fields(ast_struct: &ItemStruct) -> impl Iterator<Item = &Field> {
@@ -390,11 +396,14 @@ fn get_default(field: &Field) -> std::result::Result<Option<SqlVal>, CompilerErr
     let attr: Option<&Attribute> = field
         .attrs
         .iter()
-        .find(|attr| attr.path.is_ident("default"));
+        .find(|attr| attr.path().is_ident("default"));
     let lit: Lit = match attr {
         None => return Ok(None),
-        Some(attr) => match attr.parse_meta() {
-            Ok(Meta::NameValue(meta)) => meta.lit,
+        Some(attr) => match &attr.meta {
+            Meta::NameValue(MetaNameValue {
+                value: syn::Expr::Lit(expr_lit),
+                ..
+            }) => expr_lit.lit.clone(),
             _ => return Err(make_compile_error!("malformed default value").into()),
         },
     };
@@ -503,15 +512,18 @@ fn template_type(arguments: &syn::PathArguments) -> Option<&Ident> {
 
 fn has_derive_serialize(attrs: &[Attribute]) -> bool {
     for attr in attrs {
-        if let Ok(Meta::List(ml)) = attr.parse_meta() {
-            if ml.path.is_ident("derive")
-                && ml.nested.iter().any(|nm| match nm {
-                    NestedMeta::Meta(Meta::Path(path)) => path.is_ident("Serialize"),
-                    _ => false,
-                })
-            {
-                return true;
-            }
+        if attr.path().is_ident("derive") {
+            let mut result = false;
+            attr.parse_nested_meta(|meta| {
+                result |= meta
+                    .path
+                    .segments
+                    .iter()
+                    .any(|segment| segment.ident == "Serialize");
+                Ok(())
+            })
+            .unwrap();
+            return result;
         }
     }
     false
@@ -529,6 +541,7 @@ fn sqlval_from_lit(lit: Lit) -> std::result::Result<SqlVal, CompilerErrorMsg> {
         Lit::Verbatim(_) => {
             Err(make_compile_error!("raw verbatim literals are not supported").into())
         }
+        _ => Err(make_compile_error!("unsupported literal").into()),
     }
 }
 

